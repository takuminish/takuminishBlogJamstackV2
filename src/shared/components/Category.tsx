import Box from "@mui/material/Box";
import React from "react";
import FolderIcon from "@mui/icons-material/Folder";

type Props = {
  text: string;
};

function Category(props: Props) {
  const { text } = props;
  return (
    <Box>
      <FolderIcon
        color="primary"
        sx={{
          verticalAlign: "middle",
        }}
      />
      <Box
        component="span"
        sx={{
          verticalAlign: "middle",
          color: "#37bdf8",
          fontSize: "13px",
          fontWeight: 700,
        }}
      >
        {text}
      </Box>
    </Box>
  );
}

export default Category;
