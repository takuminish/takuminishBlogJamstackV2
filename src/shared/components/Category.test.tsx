import React from "react";
import { render, screen } from "@testing-library/react";
import Category from "./Category";

const createDOM = () => render(<Category text="test" />);

test("snapshot", () => {
  const { container } = createDOM();

  expect(container).toMatchSnapshot();
});

test("props textで与えられた文字列が表示されていること", () => {
  createDOM();

  screen.findAllByText("test");
});
