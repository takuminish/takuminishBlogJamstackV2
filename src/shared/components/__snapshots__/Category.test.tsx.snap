// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1gtxin7-MuiSvgIcon-root"
      data-testid="FolderIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
      />
    </svg>
    <span
      class="MuiBox-root css-9vtw2v"
    >
      test
    </span>
  </div>
</div>
`;
